### ISSUE TITLE
Deprecate `continue_with_*` function on `AtomisticGenericJob`

### ISSUE THREAD

**Issue by pmrv** [pyiron/pyiron_atomistics#204]
Both don't do what their names promise and imo don't add value beyond just using `restart` (in fact they poorly duplicate `restart` anyway).  Is anyone using them or otherwise has objections to me deprecating them in favor of `restart`?

See [here](https://github.com/pyiron/pyiron_atomistics/blob/66996ab6102886d4ff24c5443131a8df80f63557/pyiron_atomistics/atomistics/job/atomistic.py#L418) and [here](https://github.com/pyiron/pyiron_atomistics/blob/66996ab6102886d4ff24c5443131a8df80f63557/pyiron_atomistics/atomistics/job/atomistic.py#L437) and [restart](https://github.com/pyiron/pyiron_atomistics/blob/66996ab6102886d4ff24c5443131a8df80f63557/pyiron_atomistics/atomistics/job/atomistic.py#L394).

**Comment by jan-janssen** [comment]
No, `continue_with*` is used to define the workflow before the first calculation is finished and `restart` is used after the first calculation is finished. This functionality is used for pipelines as https://github.com/pyiron/pyiron_atomistics/blob/master/notebooks/pipeline.ipynb you can try it but I believe this notebook is going to crash if you do.

**Comment by liamhuber** [comment]
@jan-janssen @pmrv through this discussion I discovered that the inheritance tree for pipelines [is borked](https://github.com/pyiron/pyiron_base/issues/326). I tagged you both since it's related to this, but it's low enough priority to wait until next week.

**Comment by pmrv** [comment]
> No, `continue_with*` is used to define the workflow before the first calculation is finished and `restart` is used after the first calculation is finished. This functionality is used for pipelines as https://github.com/pyiron/pyiron_atomistics/blob/master/notebooks/pipeline.ipynb you can try it but I believe this notebook is going to crash if you do.

https://github.com/pyiron/pyiron_atomistics/blob/f5fd9ccda2f76b280a984f64fa5f3f7855c6c058/pyiron_atomistics/atomistics/job/atomistic.py#L418

`continue_with_restart_files()` simply calls `restart()`, saves the job, sets the old job as the parent of the new one and sets the `"structure"` field of `_generic_input` to "continue_final" (what ever that means).  Arguably we could set the parent id already in restart.

https://github.com/pyiron/pyiron_atomistics/blob/f5fd9ccda2f76b280a984f64fa5f3f7855c6c058/pyiron_atomistics/atomistics/job/atomistic.py#L437

`continue_with_final_structure()` creates a new job and sets the structure of the new job to the last structure of the old job (after it has finished), but doesn't copy any other input. 

As far as I can see this has nothing to do whether its before or after a calculation is finished, since even the `continue_*` methods call `restart` and `get_structure(-1)`. 

It's not clear to me (at all) how this relates to the pipelines, but I'll have a read.


I only bring this up, because I think restarting non-converged calculations is very common, yet picking out which method does what is non-trivial without reading the relevant code in-depth.
